python/cudf/cudf/__init__.py

```diff
@@ -32,5 +35,7 @@
 )
 from cudf.utils.utils import set_allocator
 
+cupy.cuda.set_allocator(rmm.rmm_cupy_allocator)
```

Comment:
WDYT about wrapping allocations in [`cupy.cuda.using_allocator`]( https://docs-cupy.chainer.org/en/latest/reference/generated/cupy.cuda.using_allocator.html ) contexts?

Reply:
It's honestly too widespread and I think in general if something is using cudf we want to use the same memory pool for cupy.

Reply:
We could use a utility function [analogous to the one in cuML]( https://github.com/rapidsai/cuml/blob/8242acd6ec3f1052d9ec617d3c44f20ec4166ba6/python/cuml/utils/memory_utils.py#L60 ).

Maybe but it feels a little weird from the user's standpoint if they wanted this to be some other allocator and we changed it permanently for them under-the-hood.

Reply:
The issue with that is then if a user uses cupy directly they fragment the memory space. I feel pretty strongly about making this implicit for cuDF until users voice that we should do otherwise.

Reply:
In that case, what about some messaging that indicates we are doing this under-the-hood?

Reply:
I'm happy to add some documentation (not sure exactly where) but it feels awkward to always print a warning

Reply:
Do we have any docs on cuDF and memory management already?

Reply:
> Do we have any docs on cuDF and memory management already?

As far as I know of, no. We're planning on spending significant time in 0.14 on docs so I would propose we defer this until then and I can raise an issue to document this and general memory management behavior.

Reply:
Raising an issue sounds good 👍